### PR TITLE
Updates to README focusing on using and developing the provider

### PR DESCRIPTION
**Pull request by trentrosenbaum**
The PR updates the GNUMakefile and the README.md with a focus towards onboarding a potential new developer.  Where appropriate best guess placeholders for registry documentation and naming have been used.

The local deployment has been reworked to the home `~/.terraform.d/plugins` and this should remove a need for a macos specific command.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,14 +39,55 @@ Then commit the changes to `go.mod` and `go.sum`.
...
 
-TODO - Fill this in for provider
+The RedisCloud Terraform provider is distributed through the HashiCorp managed [Terraform Registry](https://registry.terraform.io) 
+and is discovered through the use of a `required_providers`hcl block as shown in the following example.
+
+```hcl-terraform
+terraform {
+  required_providers {
+    rediscloud = {
+      source = "registry.redislabs.com/redislabs/rediscloud"
```
Comment (matpimenta) [inline README.md:49]: does it need `registry.redislabs.com` at the beginning? or can it be just `redislabs/rediscloud` for general users?

File: README.md [hunk 2]
```diff
@@ -14,9 +15,9 @@ Building The Provider
 
 1. Clone the repository
 1. Enter the repository directory
-1. Build the provider using the Go `install` command: 
+1. Build the provider using the Go `make install_macos` command:
```
Comment (matpimenta) [inline README.md:18]: Should we make this a generic (without os-specific) build information like `make` or `make build` and in the `developing the provider` section, we have the MacOS-specific instructions?

File: README.md (hunk 1 above)
Comment (matpimenta) [reply README.md:49]: Just read your note on the PR :) 
I guess we can put this and the other comments all in the `developing the provider` section.

Comment (trentrosenbaum) [reply README.md:49]: Now my understanding is we can drop the `registry.redislabs.com` and use `redislabs/rediscloud`.  I can update it to be the expected value.  I mention this in the note.

Comment (trentrosenbaum) [reply README.md]: I have reworked the flow and this example.

File: README.md (hunk 2 above)
Comment (trentrosenbaum) [reply README.md]: This is now handled through the make file and a build target.

File: GNUmakefile [hunk 3]
```diff
@@ -1,23 +1,30 @@
 default: testacc
 
 plugin_version = 99.99.99
-plugins_mirror_path = ~/Library/Application\ Support/io.terraform/plugins
-plugins_provider_path = registry.redislabs.com/redislabs/rediscloud/$(plugin_version)/darwin_amd64/
+plugins_mirror_path = ~/.terraform.d/plugins
+plugins_provider_path = registry.terraform.io/RedisLabs/rediscloud/$(plugin_version)/darwin_amd64/
```
Comment (matpimenta) [inline GNUmakefile:5]: `darwin_amd64` should probably be a variable set depending on the running environment

Comment (trentrosenbaum) [reply GNUmakefile]: Ah missed this and adjusted to make each part of the path a variable.  GOOS and GOARCH are used to populate.

**Review by matpimenta — APPROVED** [review]

**Pull request merged**